components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/AdminAdvisoryDataRetrievalClient.java

```diff
@@ -0,0 +1,134 @@
+            throws PreferenceRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            String uri = getAdminAdvisoryBannerEndpoint(tenant);
+
+            HttpGet request = new HttpGet(uri);
+            setAuthorizationHeader(request);
+
+            JSONObject jsonResponse = new JSONObject();
+
+            try (CloseableHttpResponse response = httpclient.execute(request)) {
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                    jsonResponse = new JSONObject(
+                            new JSONTokener(new InputStreamReader(response.getEntity().getContent())));
+                }
+
+                return jsonResponse;
+            } finally {
+                request.releaseConnection();
+            }
+        } catch (IOException e) {
+            String msg = "Error while getting admin advisory banner preference for tenant : " + tenant;
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(msg, e);
+            }
+
+            throw new PreferenceRetrievalClientException(msg, e);
+        }
+    }
+
+    /**
+     * Get the tenant admin advisory banner config endpoint.
+     *
+     * @param tenantDomain Tenant Domain.
+     * @return A tenant qualified endpoint.
+     * @throws PreferenceRetrievalClientException Error while getting the endpoint.
+     */
+    private String getAdminAdvisoryBannerEndpoint(String tenantDomain)
+            throws PreferenceRetrievalClientException {
+
+        return getEndpoint(tenantDomain, ADMIN_BANNER_API_RELATIVE_PATH);
+    }
+
+    /**
+     * Resolve the tenant admin advisory banner config endpoint.
+     *
+     * @param tenantDomain Tenant Domain.
+     * @param context API context Path.
+     * @return A resolved endpoint.
+     * @throws PreferenceRetrievalClientException Error while getting the base path.
+     */
+    private String getEndpoint(String tenantDomain, String context)
+            throws PreferenceRetrievalClientException {
+
+        try {
+            return IdentityManagementEndpointUtil.getBasePath(tenantDomain, context);
+        } catch (ApiException e) {
+            throw new PreferenceRetrievalClientException("Error while building url for context: " + context);
+        }
+    }
+
+    /**
+     * Set the Authorization header for a given HTTP request.
+     *
+     * @param httpMethod HTTP request method.
+     */
+    private void setAuthorizationHeader(HttpRequestBase httpMethod) {
+
+        String toEncode = IdentityManagementServiceUtil.getInstance().getAppName() + ":"
+                + String.valueOf(IdentityManagementServiceUtil.getInstance().getAppPassword());
+        byte[] encoding = Base64.encodeBase64(toEncode.getBytes());
+        String authHeader = new String(encoding, Charset.defaultCharset());
+        httpMethod.addHeader(HTTPConstants.HEADER_AUTHORIZATION, CLIENT + authHeader);
```

Comment:
Why are we using this authentication mechanism? We need to get rid of using client authentication. Isn't that an open endpoint?